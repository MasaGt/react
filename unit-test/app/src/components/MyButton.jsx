import { useState } from "react";

export const MyButton = () => {
  const [isClicked, setIsClicked] = useState(false);

  const onClick = () => {
    setIsClicked(() => {
      return !isClicked;
    });
  };

  return (
    <>
      {isClicked && <p>Button Clicked</p>}
      <button onClick={onClick}>show text</button>
    </>
  );
};
