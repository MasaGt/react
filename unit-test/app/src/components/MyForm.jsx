/**
 * フォームコンポーネント
 */

import { useState } from "react";

export const MyForm = () => {
  const [name, setName] = useState("");

  const onChange = (e) => {
    setName(e.target.value);
  };
  const isNameValid = name.length > 0;

  return (
    <>
      <form>
        <input
          type="text"
          id="name"
          value={name}
          onChange={(e) => onChange(e)}
        />
        <button type="submit" disabled={isNameValid}>
          submit
        </button>
      </form>
    </>
  );
};
