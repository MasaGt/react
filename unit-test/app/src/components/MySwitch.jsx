import { useToggle } from "../hooks/useToggle";

export const MyCacl = () => {
  const [state, change] = useToggle();

  return (
    <>
      {state && <p>You can see me</p>}
      <button onClick={change}>toggle</button>
    </>
  );
};
