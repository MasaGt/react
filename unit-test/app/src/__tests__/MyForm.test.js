/**
 * sample for snapshot test
 */

import { MyForm } from "../components/MyForm";
import { render } from "@testing-library/react";

test("snapshot sample", () => {
  const { container } = render(<MyForm />);
  expect(container).toMatchSnapshot();
});

// test("inline snapshot sample", () => {
//   render(<MyForm />);
//   // expect(container).toMatchInlineSnapshot();
//   const form = screen.getByRole("textbox");
//   expect(form).toMatchInlineSnapshot();
// });
test("inline snapshot sample", () => {
  const { container } = render(<MyForm />);

  expect(container).toMatchInlineSnapshot(`
<div>
  <form>
    <input
      id="name"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`);
});
