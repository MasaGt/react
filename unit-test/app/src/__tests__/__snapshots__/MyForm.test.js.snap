// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot sample 1`] = `
<div>
  <form>
    <input
      id="name"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`;
